### PR TITLE
Override namespace by default and only add default namespace if object does not have namespace field.

### PR DESCRIPTION
**Pull request by joonlim**

### CODE REVIEW

**Comment by joonlim** [comment]
@leeonlee @ImJasonH Could you please also give this a review?

**Comment by joonlim** [comment]
@leeonlee @ImJasonH Could you also please give this a review?

**Comment by joonlim** [comment]
@codrienne

File: gke-deploy/core/resource/resource.go [hunk 1]
```diff
@@ -234,27 +234,30 @@ func UpdateMatchingContainerImage(ctx context.Context, objs Objects, imageName,
 }
 
 // UpdateNamespace updates all objects to change its namespace to the provided namespace. Objects
-// that do not have a namespace field will not be updated.
+// that do not have a namespace field will also be updated to have a namespace field.
 func UpdateNamespace(ctx context.Context, objs Objects, replace string) error {
-	var hasNS []*Object
+	for _, obj := range objs {
+		if err := setObjectNamespace(obj, replace); err != nil {
```
Comment (codrienne) [inline gke-deploy/core/resource/resource.go]: so how does setObjectNamespace behave when the object has no namespace?

Comment (joonlim) [reply gke-deploy/core/resource/resource.go]: setObjectNamespace will set the namespace field to an object whether or not they already have one. The previous implementation checked to make sure that the field existed before setting it, but due to suggestion from product, we are now always setting the namespace in the YAML if the user provides one.

**Pull request merged**